### PR TITLE
Update 'introduce using' to support expression statements

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/37260

### CODE REVIEW

File: src/Features/CSharpTest/IntroduceUsingStatement/IntroduceUsingStatementTests.cs [hunk 1]
```diff
@@ -1417,4 +1421,40 @@ void M(System.IDisposable disposable)
...
+            }
+            """,
+            """
+            using System;
+            
+            class C
+            {
+                void M()
+                {
+                    using (MethodThatReturnsDisposableThing())
```
Comment (ToddGrun) [inline src/Features/CSharpTest/IntroduceUsingStatement/IntroduceUsingStatementTests.cs:1450]: >using (MethodThatReturnsDisposableThing()) [](http://example.com/codeflow?start=20&length=42)

Is there an option for indicating the using form desired?

using (...) {}

vs

using var _ = ... ;

File: src/Features/Core/Portable/IntroduceUsingStatement/AbstractIntroduceUsingStatementCodeRefactoringProvider.cs [hunk 2]
```diff
@@ -160,32 +192,64 @@ [.. surroundingStatements
...
+        Document document,
+        TExpressionStatementSyntax expressionStatement,
+        CancellationToken cancellationToken)
+    {
+        var surroundingStatements = GetSurroundingStatements(expressionStatement);
+        var statementIndex = surroundingStatements.IndexOf(expressionStatement);
+
+        SyntaxList<TStatementSyntax> statementsToSurround = [.. surroundingStatements.Skip(statementIndex + 1)];
+
+        var usingStatement = CreateUsingStatement(expressionStatement, statementsToSurround);
```
Comment (ToddGrun) [inline src/Features/Core/Portable/IntroduceUsingStatement/AbstractIntroduceUsingStatementCodeRefactoringProvider.cs:211]: >var usingStatement = CreateUsingStatement(expressionStatement, statementsToSurround); [](http://example.com/codeflow?start=8&length=85)

nit: consider moving inside ReplaceWithUsingStatementAsync

File: src/Features/CSharpTest/IntroduceUsingStatement/IntroduceUsingStatementTests.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/Features/CSharpTest/IntroduceUsingStatement/IntroduceUsingStatementTests.cs:1464]: done

File: src/Features/Core/Portable/IntroduceUsingStatement/AbstractIntroduceUsingStatementCodeRefactoringProvider.cs [hunk 3]
```diff
@@ -57,42 +59,72 @@ public override async Task ComputeRefactoringsAsync(CodeRefactoringContext conte
...
+                    CodeActionTitle,
+                    cancellationToken => IntroduceUsingStatementAsync(document, expressionStatement, cancellationToken),
+                    CodeActionTitle),
+                expressionStatement.Span);
+        }
+    }
+
+    private async Task<TExpressionStatementSyntax?> FindDisposableExpressionStatementAsync(Document document, TextSpan selection, CancellationToken cancellationToken)
+    {
+        var expressionStatement = await document.TryGetRelevantNodeAsync<TExpressionStatementSyntax>(selection, cancellationToken).ConfigureAwait(false);
```
[the remark this answers is not in the document]
Comment (CyrusNajmabadi) [reply src/Features/Core/Portable/IntroduceUsingStatement/AbstractIntroduceUsingStatementCodeRefactoringProvider.cs]: done.

File: src/Features/Core/Portable/IntroduceUsingStatement/AbstractIntroduceUsingStatementCodeRefactoringProvider.cs [hunk 4]
```diff
@@ -160,32 +192,64 @@ [.. surroundingStatements
...
+        Document document,
+        TStatementSyntax statementToReplace,
+        SyntaxList<TStatementSyntax> statementsToSurround,
+        TStatementSyntax usingStatement,
+        CancellationToken cancellationToken)
+    {
+        var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+        var surroundingStatements = GetSurroundingStatements(statementToReplace);
+        var statementIndex = surroundingStatements.IndexOf(statementToReplace);
```
[the remark this answers is not in the document]
Comment (CyrusNajmabadi) [reply src/Features/Core/Portable/IntroduceUsingStatement/AbstractIntroduceUsingStatementCodeRefactoringProvider.cs]: done.

File: src/Features/Core/Portable/IntroduceUsingStatement/AbstractIntroduceUsingStatementCodeRefactoringProvider.cs (hunk 2 above)
Comment (CyrusNajmabadi) [reply src/Features/Core/Portable/IntroduceUsingStatement/AbstractIntroduceUsingStatementCodeRefactoringProvider.cs]: Different types of using statements are created (i updated their names to make this clearer).

File: src/Features/Core/Portable/IntroduceUsingStatement/AbstractIntroduceUsingStatementCodeRefactoringProvider.cs [hunk 5]
```diff
@@ -11,100 +11,136 @@
...
+        var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
         var disposableType = semanticModel.Compilation.GetSpecialType(SpecialType.System_IDisposable);
         if (disposableType is null)
-            return default;
+            return;
 
-        var operation = semanticModel.GetOperation(declarationSyntax, cancellationToken) as IVariableDeclarationGroupOperation;
-        if (operation?.Declarations.Length != 1)
-            return default;
+        var syntaxFacts = document.GetRequiredLanguageService<ISyntaxFactsService>();
```
Comment (ToddGrun) [inline src/Features/Core/Portable/IntroduceUsingStatement/AbstractIntroduceUsingStatementCodeRefactoringProvider.cs:72]: > var syntaxFacts = document.GetRequiredLanguageService<ISyntaxFactsService>(); [](http://example.com/codeflow?start=7&length=78)

super nit: move inside else clause

**Review by ToddGrun — APPROVED** [review]
:shipit:

**Pull request merged**